shell/common/engine.cc

```diff
@@ -403,35 +401,15 @@ bool Engine::HandleLocalizationPlatformMessage(
   return true;
 }
 
-bool Engine::HandleSystemPlatformMessage(blink::PlatformMessage* message) {
+void Engine::HandleSettingsPlatformMessage(blink::PlatformMessage* message) {
   const auto& data = message->data();
-  rapidjson::Document document;
-  document.Parse(reinterpret_cast<const char*>(data.data()), data.size());
-
-  if (document.HasParseError() || !document.IsObject())
-    return false;
-
-  auto root = document.GetObject();
-  auto type = root.FindMember("type");
-  if (type == root.MemberEnd() || type->value != "systemSettings")
-    return false;
-
-  // This only handles textScaleFactor changes: other system messages
-  // are handled by DispatchPlatformMessage.
-  auto text_scale_factor = root.FindMember("textScaleFactor");
-  if (text_scale_factor == root.MemberEnd() ||
-      !text_scale_factor->value.IsDouble()) {
-    return false;
-  }
-  text_scale_factor_ = text_scale_factor->value.GetDouble();
+  std::string jsonData(reinterpret_cast<const char*>(data.data()), data.size());
+  user_settings_data_ = jsonData;
   if (runtime_) {
-    runtime_->SetTextScaleFactor(text_scale_factor_);
+    runtime_->SetUserSettingsData(user_settings_data_);
```

Comment:
Please keep the setters separate. Type information is now lost.

Reply:
We don't need this type information in the C++ code. C++ just passes it through. I'm changing it so that C++ doesn't interpret the contents at all. Otherwise, next time someone needs to add another user setting field, they will have to touch 9 files, if not more. On top of that, the engine contains some funky logic that attempts to synchronize Dart runtime initialization with passing fields (see the `if (runtime_)` checks in both `engine.cc`, `if (!dart_state)` in `window.cc`, and field caches in `engine.h` and `runtime_controller.h`, such as `text_scale_factor_`). To apply this logic on every individual field seems like overkill.